legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/ProjectsServiceImpl.java

```diff
@@ -301,20 +303,38 @@ public ProjectDependencyReport buildReportFromGraph(ProjectDependencyGraph depen
     }
 
     @Override
-    public List<ProjectDependencyWithPlatformVersions> getDependentProjects(String groupId, String artifactId, String versionId)
+    public List<ProjectDependencyWithPlatformVersions> getDependantProjects(String groupId, String artifactId, String versionId, boolean latestOnly)
     {
+        List<ProjectDependencyWithPlatformVersions> result;
         if (versionId.equalsIgnoreCase("ALL"))
         {
-            return projectsVersions.getAll().stream().map(projectData -> projectData.getVersionData().getDependencies().stream()
+            result = projectsVersions.getAll().stream().map(projectData -> projectData.getVersionData().getDependencies().stream()
                     .filter(dep -> dep.getGroupId().equals(groupId) && dep.getArtifactId().equals(artifactId))
                     .map(dep -> new ProjectDependencyWithPlatformVersions(projectData.getGroupId(), projectData.getArtifactId(), projectData.getVersionId(), dep,projectData.getVersionData().getProperties()))
                     .collect(Collectors.toList())).flatMap(Collection::stream).collect(Collectors.toList());
         }
-        String version =  this.resolveAliasesAndCheckVersionExists(groupId, artifactId, versionId);
-        return projectsVersions.getAll().stream().map(projectData -> projectData.getVersionData().getDependencies().stream()
-                .filter(dep -> dep.getGroupId().equals(groupId) && dep.getArtifactId().equals(artifactId) && dep.getVersionId().equals(version))
-                .map(dep -> new ProjectDependencyWithPlatformVersions(projectData.getGroupId(), projectData.getArtifactId(), projectData.getVersionId(), dep,projectData.getVersionData().getProperties()))
-                .collect(Collectors.toList())).flatMap(Collection::stream).collect(Collectors.toList());
+        else
+        {
+            String version =  this.resolveAliasesAndCheckVersionExists(groupId, artifactId, versionId);
+            result = projectsVersions.getAll().stream().map(projectData -> projectData.getVersionData().getDependencies().stream()
+                    .filter(dep -> dep.getGroupId().equals(groupId) && dep.getArtifactId().equals(artifactId) && dep.getVersionId().equals(version))
+                    .map(dep -> new ProjectDependencyWithPlatformVersions(projectData.getGroupId(), projectData.getArtifactId(), projectData.getVersionId(), dep,projectData.getVersionData().getProperties()))
+                    .collect(Collectors.toList())).flatMap(Collection::stream).collect(Collectors.toList());
+        }
+        if (latestOnly)
```

Comment:
you can move this to a new function for better readability , maybe call it filterResultBYLatest (just a suggestion)